src/RepositoryAPI.jl

```diff
@@ -456,9 +137,151 @@ end
 
 
 
+# Internal convenience function.  Updates an object from the repository.
+function postRepositoryObject(token::AbstractString,
+                              objectType::AbstractString,
+                              searchKey::Dict{Symbol, Any};
+                              attributes::Dict=Dict(),
+                              objectFields::Dict=Dict(),
+                              filterRequired::Bool=true
+)
+
+    global verbose
+
+    if token == ""
+        throw(ArgumentError("`token' cannot be empty"))
+    end
+
+    objectID = get(searchKey, :id, 0)
+    name = get(searchKey, :name, "")
+
+    # If objectID is not supplied, retrieve from get function
+    if objectID == 0 
+        if objectType == "alert"
+            object = getRepositoryAlert(token, alertName = name)
+            objectID = get(object, "id", 0)
+        elseif objectType == "domain"
+            object = getRepositoryDomain(token, appName = name)
+            objectID = get(object, "id", 0)
+        elseif objectType == "tenant"
+            object = getRepositoryTenant(token, name = name)
+            objectID = get(object, "id", 0)
+        end
+    end
+
+    local isKeySet = false
+
+    if filterRequired
+        isKeySet = any(kv -> isa(kv[2], Number) ? kv[2] > 0 : !isempty(kv[2]), searchKey)
+
+        if !isKeySet
+            throw(ArgumentError("At least one of `$(join(collect(keys(searchKey)), "', `", "' or `"))' must be set"))
+        end
+    end
+
+    local url = ObjectEndpoint * "/" * objectType * "/$(objectID)"
+    local debugID = "(all)"
+
+    if verbose
+        println("POST $url")
+        println("X-Auth-Token: $token")
+        println(attributes)
+    end
+
+
+    json = Dict{AbstractString, Any}()
+    json["type"] = objectType
+
+    # If attributes is supplied, update the object’s attributes field
+    if !isempty(attributes)
+        attributesDict = []
+
+        for (key, val) in attributes
+            push!(attributesDict, Dict("name" => key, "value"=> val))
+        end
+        
+        json["attributes"] = attributesDict
+    end
+
+    # If any objectFields are supplied by the user, update these in the object (if it exists)
+    if !isempty(objectFields)
+        for (key, val) in objectFields
```

Comment:
what if one of the `key`s is `attributes`?  Will that overwrite attributes above?

Reply:
The way it is currently, yes `attributes` would be overwritten.  Should we check for `attributes` (and remove) if `objectFields` has an entry for this?  

Prior to this, we can add logic to determine how to choose `attributes`?

Reply:
You can just move the attributes setting block to after this block.

Reply:
ah, that works.  Moved it after this block